lib_gb/src/ppu/fifo/fifo_ppu.rs

```diff
@@ -0,0 +1,301 @@
+                                        self.state = PpuState::Hblank;
+                                        if self.h_blank_interrupt_request{
+                                            self.trigger_stat_interrupt = true;
+                                        }
+                                        self.bg_fetcher.try_increment_window_counter();
+                                        self.bg_fetcher.reset();
+                                        self.sprite_fetcher.reset();
+
+                                        // If im on the first iteration and finished the 160 pixels break;
+                                        // In this case the number of t_cycles should be eneven but it will break
+                                        // my code way too much for now so Im leaving this as it is... (maybe in the future)
+                                        break;
+                                    }
+                            }
+                        }
+                    }
+                
+
+                    
+                    self.t_cycles_passed += 2;
+                }
+            }
+            
+        }
+    }
+
+    fn try_push_to_lcd(&mut self){
+        if !self.bg_fetcher.fifo.is_empty(){
+            if !self.scanline_started{
+                // discard the next pixel in the bg fifo
+                // the bg fifo should start with 8 pixels and not push more untill its empty again
+                if 8 - self.bg_fetcher.fifo.len() >= self.bg_pos.x as usize % 8{
+                    self.scanline_started = true;
+                }
+                else{
+                    self.bg_fetcher.fifo.remove(0);
+                    return;
+                }
+            }
+
+            let bg_pixel_color_num = self.bg_fetcher.fifo.remove(0);
+            let bg_pixel = self.bg_color_mapping[bg_pixel_color_num as usize];
+            let pixel = if !self.sprite_fetcher.fifo.is_empty(){
+                let sprite_color_num = self.sprite_fetcher.fifo.remove(0);
+                let pixel_oam_attribute = &self.sprite_fetcher.oam_entries[sprite_color_num.1 as usize];
+
+                if sprite_color_num.0 == 0 || (pixel_oam_attribute.is_bg_priority && bg_pixel_color_num != 0){
+                    bg_pixel
+                }
+                else{
+                    let sprite_pixel = if pixel_oam_attribute.palette_number{
+                        self.obj_color_mapping1[sprite_color_num.0 as usize]
+                    }
+                    else{
+                        self.obj_color_mapping0[sprite_color_num.0 as usize]
+                    };
+
+                    sprite_pixel.expect("Corruption in the object color pallete")
+                }
+            }
+            else{
+                bg_pixel
+            };
+
+            self.push_lcd_buffer.push(pixel);
+            self.pixel_x_pos += 1;
+        }
+    }
+
+    fn color_as_uint(color: &Color) -> u32 {
```

Comment:
Maybe implement inside `Color` and implement the trait [Into<T>](https://doc.rust-lang.org/std/convert/trait.Into.html)